### PR TITLE
go/runtime/host/sgx: Autodetect SGX device name

### LINKED ISSUE #4333: Discover SGX driver device location
It seems that some later drivers use `/dev/sgx` instead of `/dev/isgx` but currently Oasis Core hardcodes the path as `/dev/isgx` so people need to create symlinks.

There should be some driver autodetection logic which would bind the supported device into the sandbox.

### PR DESCRIPTION
**Pull request by abukosek**
Closes #4333.

### CODE REVIEW

File: go/runtime/host/sgx/sgx.go [hunk 1]
```diff
@@ -145,6 +146,23 @@ func (s *sgxProvisioner) loadEnclaveBinaries(rtCfg host.Config) ([]byte, []byte,
 	return sgxs, sig, nil
 }
 
+func (s *sgxProvisioner) discoverSGXDevice() (string, error) {
+	// Different versions of Intel SGX drivers provide different names for
+	// the SGX device.  Autodetect which one actually exists.
+	sgxDevices := []string{"/dev/sgx", "/dev/isgx"}
```
Comment (kostko) [inline go/runtime/host/sgx/sgx.go:152]: There's also a `/dev/sgx/enclave` variant I think.

Comment (abukosek) [reply go/runtime/host/sgx/sgx.go]: Thanks, added!

File: go/runtime/host/sgx/sgx.go [hunk 2]
```diff
@@ -145,6 +146,23 @@ func (s *sgxProvisioner) loadEnclaveBinaries(rtCfg host.Config) ([]byte, []byte,
 	return sgxs, sig, nil
 }
 
+func (s *sgxProvisioner) discoverSGXDevice() (string, error) {
+	// Different versions of Intel SGX drivers provide different names for
+	// the SGX device.  Autodetect which one actually exists.
+	sgxDevices := []string{"/dev/sgx", "/dev/sgx/enclave", "/dev/isgx"}
```
Comment (kostko) [inline go/runtime/host/sgx/sgx.go:152]: And maybe also `/dev/sgx_enclave` (see https://github.com/fortanix/rust-sgx/blob/master/intel-sgx/sgxs-loaders/src/isgx/mod.rs#L444).

**Review by kostko — APPROVED** [review]
Also please backport to the `stable/21.3.x` branch.

**Pull request merged**